Gems/Atom/RHI/Vulkan/Code/Source/RHI/DescriptorSetLayout.cpp

```diff
@@ -159,6 +159,30 @@ namespace AZ
             const AZStd::span<const RHI::ShaderInputSamplerDescriptor> samplerDescs = m_shaderResourceGroupLayout->GetShaderInputListForSamplers();
             const AZStd::span<const RHI::ShaderInputStaticSamplerDescriptor>& staticSamplerDescs = m_shaderResourceGroupLayout->GetStaticSamplers();
 
+            // About VK_SHADER_STAGE_ALL...
+            // We attempted to configure the descriptor set with the actual resource visibility but it was problematic...
+            // Vulkan requires that the visibility flags used to create the VkDescriptorSet (RHI::ShaderResourceGroup) must exactly match the ones
+            // used to create the VkPipelineLayout (RHI::PipelineState).
+            // See https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkCmdBindDescriptorSets-pDescriptorSets-00358
+            // But Atom currently expects to be able to use certain ShaderResourceGroup instances with many different pipeline states regardless of visibility.
+            // - ShaderResourceGroupLayouts for "SceneSrg" and "ViewSrg" are defined in a special SceneAndViewSrgs.shader file. This shader has no entry points
+            //   and the asset only exists to provide these SRG layouts. The runtime loads this special shader and uses it to instantiate the one
+            //   SceneSrg (and ViewSrg(s)) and uses this instance for many shaders with different resource visibilities.
+            // - Same for RayTracingSrgs.shader's "RayTracingSceneSrg" and "RayTracingMaterialSrg"
+            // - Same for ForwardPassSrg.shader's "PassSrg". (This one is especially problematic because, unlike the above cases, we can't just add some special
+            //   handling for the particular SRG name; "PassSrg" is widely used as the name for many different per-pass SRG layouts).
+            // - ShaderResourceGroupPool is intentionally set up to reuse SRGs regardless of visibility, per ShaderResourceGroup::MakeInstanceId which uses
+            //   the source file path in the unique ID (ex: "D:\o3de\Gems\Atom\RPI\Assets\ShaderLib\Atom\RPI\ShaderResourceGroups\DefaultDrawSrg.azsli")
+            //   so that any shader that uses this azsli file will share the same pool and thus share the same PipelineLayoutDescriptor.
+            // In order to address the above issues, we would need to update AZSLc to support some kind of attribute by which the shader-author can manually
+            // override the visibility for each resource, or we add some new metadata to the .shader files to provide explicit overrides for particular resource
+            // visibilities. Either way, this would likely become error prone and difficult to maintain.
+            // 
+            // There are even more issues beyond what's described above; I prototyped some code to address them and I was still getting RHI validation
+            // failures. I had to give up and move on, so I don't know how deep this rabbit hole goes.
```

Comment:
I would remove this comment.  Instead consider saying

"It is possible that even with the proposed solution above we could run into other issues not foreseen here. Instead a better path forward would be to see if Khronos can address the api requirement itself. A ticket has been opened with Khronos here - https://github.com/KhronosGroup/Vulkan-Docs/issues/1790.
For context the driver error when the visibility bit from PSO does not match the one from Descriptor set is as follows ->